helpers/stable_swap_math/mod.rs

```diff
@@ -320,8 +316,10 @@ pub fn rated_swap_from(
         fees,
         amp_coef,
     )?;
-
-    let dy = amount_from_rated(r_dy, rates[token_in_idx])?;
+    // add one in case of rounding error, for the protocol advantage
+    let dy = amount_from_rated(r_dy, rates[token_in_idx])?
+        .checked_add(1)
```

Comment:
As you note in the PR description – isn't this leaking 1 token out of the protocol? i.e. (very very) slowly draining the pool?

Reply:
This change does the opposite. It forces that the protocol never gets less "token_in" than in the `rated_swap_to(...)` for the same (or slightly smaller*) amount of "token_out".

*in case when "token_out" has smaller precision than "token_in"

Reply:
Ah, b/c it's used to calculate how much of `token_in` the traders has to provide in order to receive `token_out_amount` 🙈 

See, I was confused b/c it's called in [`get_swap_amount_in`](https://github.com/Cardinal-Cryptography/common-amm-stable-swap/blob/main/amm/contracts/stable_pool/lib.rs#L887) which made it sound like swapping _given_ some token_in....